### ISSUE TITLE
Add support for primitive types in getters and setters of non-nullable fields

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#74]
The non-nullability can be expressed in Java by using int, double, float, long, etc instead of Integer, Double, Float, Long...

---

This feature is rejected. There is only little gain from using primitive types

**Comment by perlun** [comment]
Coming into this some 10 years later, I stumbled across it since I started noticing that even though my DB table fields are created as `NOT NULL`, the backing fields & getters in my immutable POJO class uses `Integer`, `Boolean` and so forth instead of the more "natural" (for non-nullable fields) `int`, `boolean` types.

I guess it's the same thing as usual though: no-one else has asked for it in the (almost) 10 years since this issue was originally crated? I must say I feel truly unique... :joy:

**Comment by lukaseder** [comment]
No, this time, you're looking at the most popular feature request (in kotlin): https://github.com/jOOQ/jOOQ/issues/10212

TL;DR: `NOT NULL` guarantees are a lie, and you probably don't *really* want that
